### PR TITLE
docs: add installation steps of tools for developers

### PR DESCRIPTION
**Pull request by kenji-miyake**

### CODE REVIEW

File: docs/installation/tools-for-developers/index.md [hunk 1]
```diff
@@ -1,5 +1,32 @@
...
+## ROS 2 settings
 
-    Under Construction
+### Colorizing logger output
+
+By default, ROS 2 logger doesn't colorize the output.
+To colorize it, write the following in your `.bashrc`:
+
+```bash
+export RCUTILS_COLORIZED_OUTPUT=1
```
Comment (xmfcx) [inline docs/installation/tools-for-developers/index.md:11]: Wow I didn't know this, thanks!

File: docs/installation/tools-for-developers/index.md [hunk 2]
```diff
@@ -1,5 +1,32 @@
...
+
+```bash
+export RCUTILS_CONSOLE_OUTPUT_FORMAT="[{severity} {time}] [{name}]: {message} ({function_name}() at {file_name}:{line_number})"
+```
+
+See [here](https://docs.ros.org/en/rolling/Tutorials/Logging-and-logger-configuration.html#console-output-formatting) for more options.
+
+### Enabling localhost-only communication
+
+By default, ROS 2 communicates using multi-cast, which may unnecessaryly increase the network traffic.
```
[the remark this answers is not in the document]
Comment (kenji-miyake) [reply docs/installation/tools-for-developers/index.md:27]: Will check the settings of the dictionary.

[the remark this answers is not in the document]
Comment (kenji-miyake) [reply docs/installation/tools-for-developers/index.md]: Since I've changed the workflow name, there was no workflow file at a certain point.
This PR was created at the timing. :pleading_face:

**Review by xmfcx — DISMISSED** [review]

**Comment by kenji-miyake** [comment]
I'll merge this after https://github.com/autowarefoundation/autoware-documentation/pull/18.

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+# JetBrains IDEs
+/.idea/
+
+# VS Code
+.vscode/*
+!.vscode/settings.json
+!.vscode/tasks.json
+!.vscode/launch.json
+!.vscode/extensions.json
+!.vscode/*.code-snippets
```
Comment (kenji-miyake) [inline .gitignore:11]: @xmfcx Do you want to add these json and snippets files?
If not, I think we can simplify this.

Also, can I make another PR for `.gitignore` to make the diff easy to understand?

File: docs/installation/tools-for-developers/index.md [hunk 4]
```diff
@@ -1,5 +1,51 @@
...
+### Setting up `ROS_DOMAIN_ID`
+
+ROS 2 nodes on the same domain can freely discover and send messages to each
+other, while ROS 2 nodes on different domains cannot. All ROS 2 nodes use
+domain ID 0 by default. To avoid interference between different groups of
+computers running ROS 2 on the same network, a different domain ID should
+be set for each group.
+
+See [here](https://docs.ros.org/en/foxy/Concepts/About-Domain-ID.html#the-ros-domain-id)
+for more information.
```
Comment (kenji-miyake) [inline docs/installation/tools-for-developers/index.md:43]: I'd like to discuss the style of Markdown files.
Since Markdown will automatically set the layout, I think we don't have to adjust it manually.

The disadvantages of writing like this are:
- The positions of period marks become unclear.
- People have to adjust the layout when they edit it.

If it's `DISCLAIMER` like [this](https://gitlab.com/autowarefoundation/autoware.auto/AutowareAuto/-/blob/master/DISCLAIMER.md), people might read it in the raw format, it's fine to adjust the layout.
But I believe we don't need it for Markdown files.

What do you think about it? @xmfcx

File: .gitignore (hunk 3 above)
Comment (kenji-miyake) [reply .gitignore]: https://github.com/autowarefoundation/autoware-documentation/pull/30

File: docs/installation/tools-for-developers/index.md (hunk 4 above)
Comment (kenji-miyake) [reply docs/installation/tools-for-developers/index.md]: I've simplified it, how about it? dd12560c2c44e8114cf50ce3ef1b08e89cddd858

File: docs/installation/autoware/source-installation.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+   cd autoware
+   ```
+
+2. Install the dependencies.
+
+   We use [Ansible](https://www.ansible.com/) to simplify the steps.
+   Please see the Ansible playbooks and roles if you want to know exactly what packages are installed.
+
+   > Note: Before installing NVIDIA libraries, confirm and agree with the licenses.
+
```
Comment (xmfcx) [inline docs/installation/autoware/source-installation.md:29]:
```suggestion

> Be very careful with this method. Make sure you read and confirmed all the steps in the ansible configuration before using it.

```

**Review by xmfcx — CHANGES_REQUESTED** [review]

File: docs/installation/autoware/source-installation.md (hunk 5 above)
Comment (kenji-miyake) [reply docs/installation/autoware/source-installation.md:29]: I'll use `!!!warning` instead!

File: docs/installation/tools-for-developers/index.md (hunk 4 above)
Comment (xmfcx) [reply docs/installation/tools-for-developers/index.md]: @kenji-miyake why not use something like https://github.com/prettier/prettier for formatting markdown files automatically?

File: docs/installation/autoware/source-installation.md (hunk 5 above)
Comment (kenji-miyake) [reply docs/installation/autoware/source-installation.md]: Actually, this was shown because the branch was old.
But anyway I'll fix this.

Comment (kenji-miyake) [reply docs/installation/autoware/source-installation.md]: @xmfcx a061ddea82ff0789e1e96b5504e60155c66b2627
image

Comment (xmfcx) [reply docs/installation/autoware/source-installation.md]: Can you pıt the warning before the installation command?

Comment (kenji-miyake) [reply docs/installation/autoware/source-installation.md]: Unfortunately, it was impossible to write it in the indented section because of the constraints of `Prettier` and `markdownlint`. :pleading_face: 

image

Comment (xmfcx) [reply docs/installation/autoware/source-installation.md]: Ah ok.

Comment (kenji-miyake) [reply docs/installation/autoware/source-installation.md]: Also, it wasn't displayed correctly.
image

File: docs/installation/tools-for-developers/index.md (hunk 4 above)
Comment (kenji-miyake) [reply docs/installation/tools-for-developers/index.md]: @xmfcx I already use it, but I guess this isn't a supported use case.

Comment (kenji-miyake) [reply docs/installation/tools-for-developers/index.md]: FYI: It's automatically checked using `pre-commit`.

https://github.com/autowarefoundation/autoware-documentation/runs/5513786672?check_suite_focus=true#step:3:112
https://github.com/autowarefoundation/autoware-documentation/blob/41eeea7b6be1ffbd92fed2e4bc892437a7db88d3/.pre-commit-config.yaml#L27-L30

Comment (xmfcx) [reply docs/installation/tools-for-developers/index.md]: I get it, I'm ok with your choice on this.

**Review by xmfcx — APPROVED** [review]
